kuma/wiki/tests/test_views_list.py

```diff
@@ -121,3 +145,201 @@ def test_revisions_request_invalid_pages(root_doc, client):
+def test_list_with_errors(redirect_doc, doc_hierarchy_with_zones, client,
+                          locale):
+    top_doc = doc_hierarchy_with_zones.top
+    bottom_doc = doc_hierarchy_with_zones.bottom
+    de_doc = top_doc.translated_to('de')
+    for doc in (top_doc, bottom_doc, de_doc, redirect_doc):
+        doc.rendered_errors = 'bad render'
+        doc.save()
+
+    if locale == 'en-US':
+        exp_docs = (top_doc, bottom_doc)
+    elif locale == 'de':
+        exp_docs = (de_doc,)
+    else:  # fr
+        exp_docs = ()
+
+    url = reverse('wiki.errors', locale=locale)
+    resp = client.get(url)
+    dom = pq(resp.content)
+    assert resp.status_code == 200
+    assert 'public' in resp['Cache-Control']
+    assert 's-maxage' in resp['Cache-Control']
+    assert 'text/html' in resp['Content-Type']
+    assert len(dom.find('.document-list li')) == len(exp_docs)
+    selector = 'ul.document-list li a[href="/{}/docs/{}"]'
+    for doc in exp_docs:
+        assert len(dom.find(selector.format(doc.locale, doc.slug))) == 1
+
+
+@pytest.mark.parametrize('locale', ['en-US', 'de', 'fr'])
+def test_list_without_parent(redirect_doc, doc_hierarchy_with_zones, client,
+                             locale):
+    if locale == 'en-US':
+        exp_docs = (doc_hierarchy_with_zones.top,
+                    doc_hierarchy_with_zones.middle_top,
+                    doc_hierarchy_with_zones.middle_bottom,
+                    doc_hierarchy_with_zones.bottom)
+    else:  # All translations have a parent.
+        exp_docs = ()
+
+    url = reverse('wiki.without_parent', locale=locale)
+    resp = client.get(url)
+    dom = pq(resp.content)
+    assert resp.status_code == 200
+    assert 'public' in resp['Cache-Control']
+    assert 's-maxage' in resp['Cache-Control']
+    assert 'text/html' in resp['Content-Type']
+    assert len(dom.find('.document-list li')) == len(exp_docs)
+    selector = 'ul.document-list li a[href="/{}/docs/{}"]'
+    for doc in exp_docs:
+        assert len(dom.find(selector.format(doc.locale, doc.slug))) == 1
+
+
+@pytest.mark.parametrize('locale', ['en-US', 'de', 'fr'])
+def test_list_top_level(redirect_doc, doc_hierarchy_with_zones, client,
+                        locale):
+    if locale == 'en-US':
+        exp_docs = (doc_hierarchy_with_zones.top,)
+    else:
+        exp_docs = (doc_hierarchy_with_zones.top.translated_to(locale),)
+
+    url = reverse('wiki.top_level', locale=locale)
+    resp = client.get(url)
+    dom = pq(resp.content)
+    assert resp.status_code == 200
+    assert 'public' in resp['Cache-Control']
+    assert 's-maxage' in resp['Cache-Control']
+    assert 'text/html' in resp['Content-Type']
+    assert len(dom.find('.document-list li')) == len(exp_docs)
+    selector = 'ul.document-list li a[href="/{}/docs/{}"]'
+    for doc in exp_docs:
+        assert len(dom.find(selector.format(doc.locale, doc.slug))) == 1
+
+
+@pytest.mark.parametrize('tag', ['foo', 'bar'])
```

Comment:
Nit: I don't think parametrization should be used here. Localization tags are less flexible than they seem from looking at the code. There's exactly one, "inprogress". It's unclear why this was implemented as a tag instead of a boolean on Revisions.

Reply:
Thanks, that's good to know.